File: ABC304/d.py
w, h = map(int, input().split())
n = int(input())

fruits_pos = [tuple(map(int, input().split())) for _ in range(n)]

a = int(input())
cut_y = [0]
cut_y += list(map(int, input().split()))
cut_y += [w]
cut_y_pos =[]
for i in range(len(cut_y) -1):
    cut_y_pos.append(tuple([cut_y[i], cut_y[i+1]]))

b = int(input())
cut_x = [0]
cut_x += list(map(int, input().split()))
cut_x += [h]
cut_x_pos =[]
for i in range(len(cut_x) -1):
    cut_x_pos.append(tuple([cut_x[i], cut_x[i+1]]))

cur_min = 0
cur_max = 0
for pos_x in cut_x_pos:
    for pos_y in cut_y_pos:
        tmp = 0
        for fruit in fruits_pos:
            if pos_y[0] < fruit[0] < pos_y[1] and pos_x[0] < fruit[1] < pos_x[1]:
                tmp += 1
        cut_min = min(cur_min, tmp)
        cur_max = max(cur_max, tmp)
print("{} {}".format(cur_min, cur_max))

File: ABC304/d.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int w, h, n, a, b;
    cin >> w >> h;
    cin >> n;

    vector<pair<int, int>> fruits_pos(n);
    for (int i = 0; i < n; ++i) {
        cin >> fruits_pos[i].first >> fruits_pos[i].second;
    }

    cin >> a;
    vector<int> cut_y(a+2);
    cut_y[0] = 0;
    for (int i = 1; i <= a; ++i) {
        cin >> cut_y[i];
    }
    cut_y[a+1] = w;

    vector<pair<int, int>> cut_y_pos(a+1);
    for (int i = 0; i < a+1; ++i) {
        cut_y_pos[i] = make_pair(cut_y[i], cut_y[i+1]);
    }

    cin >> b;
    vector<int> cut_x(b+2);
    cut_x[0] = 0;
    for (int i = 1; i <= b; ++i) {
        cin >> cut_x[i];
    }
    cut_x[b+1] = h;

    vector<pair<int, int>> cut_x_pos(b+1);
    for (int i = 0; i < b+1; ++i) {
        cut_x_pos[i] = make_pair(cut_x[i], cut_x[i+1]);
    }

    for (auto &pos: cut_x_pos) {
        cout << pos.first << ' ' << pos.second << endl;
    }

    vector<int> num_of_fruits;
    for (auto &pos_x: cut_x_pos) {
        for (auto &pos_y: cut_y_pos) {
            int tmp = 0;
            for (auto &fruit: fruits_pos) {
                if (pos_y.first < fruit.first && fruit.first < pos_y.second 
                && pos_x.first < fruit.second && fruit.second < pos_x.second) {
                    ++tmp;
                }
            }
            num_of_fruits.push_back(tmp);
        }
    }

    auto minmax = minmax_element(num_of_fruits.begin(), num_of_fruits.end());
    cout << *minmax.first << ' ' << *minmax.second << endl;

    return 0;
}
